app-modules/interaction/src/Models/Interaction.php

```diff
@@ -0,0 +1,180 @@
+ * @method static \Illuminate\Database\Eloquent\Builder|Interaction whereCreatedAt($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|Interaction whereDescription($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|Interaction whereEndDatetime($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|Interaction whereId($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|Interaction whereInteractableId($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|Interaction whereInteractableType($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|Interaction whereInteractionCampaignId($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|Interaction whereInteractionDriverId($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|Interaction whereInteractionInstitutionId($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|Interaction whereInteractionOutcomeId($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|Interaction whereInteractionRelationId($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|Interaction whereInteractionStatusId($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|Interaction whereInteractionTypeId($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|Interaction whereStartDatetime($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|Interaction whereSubject($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|Interaction whereUpdatedAt($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|Interaction whereUserId($value)
+ *
+ * @mixin \Eloquent
+ */
+class Interaction extends BaseModel implements Auditable, CanTriggerAutoSubscription
+{
+    use AuditableTrait;
+
+    protected $fillable = [
+        'user_id',
+        'interactable_id',
+        'interactable_type',
+        'interaction_campaign_id',
+        'interaction_driver_id',
+        'interaction_institution_id',
+        'interaction_outcome_id',
+        'interaction_relation_id',
+        'interaction_status_id',
+        'interaction_type_id',
+        'start_datetime',
+        'end_datetime',
+        'subject',
+        'description',
+    ];
+
+    protected $casts = [
+        'start_datetime' => 'datetime',
+        'end_datetime' => 'datetime',
+    ];
+
+    public function getSubscribable(): ?Subscribable
+    {
+        return $this->interactable instanceof Subscribable ? $this->interactable : null;
+    }
+
+    public function interactable(): MorphTo
+    {
+        return $this->morphTo(
+            name: 'interactable',
+            type: 'interactable_type',
+            id: 'interactable_id',
+        );
+    }
+
+    public function interactionCampaign(): BelongsTo
+    {
+        return $this->belongsTo(InteractionCampaign::class);
+    }
+
+    public function campaign(): BelongsTo
+    {
+        return $this->interactionCampaign();
+    }
```

Comment:
Rather than define two different methods, it may be cleaner to define the `relation` parameter like so:

```
public function campaign(): BelongsTo
    {
        return $this->interactionCampaign(relation: interactionCampaign);
    }
```
Removing the need for the `interactionCampaign` method.

Or is there a different reason to define the two methods?

The same goes for the other relations here.

Reply:
Sounds good - will make the adjustment here and add a backlog item for Engagement.